mesh/node.c

```diff
@@ -1672,13 +2040,108 @@ void node_join(const char *app_path, const char *sender, const uint8_t *uuid,
 					req, l_free);
 }
 
+void node_import(const char *sender, void *json_data, const uint8_t *uuid,
+			node_ready_func_t cb, void *user_data)
+{
+	struct mesh_node *node;
+	struct mesh_net *net;
+	struct mesh_db_node db_node;
+	struct keyring_net_key *net_keyring;
+	struct l_queue *net_keys;
+
+	json_object *jnode = json_data;
+
+	uint16_t unicast;
+	uint8_t dev_key[16];
+	uint32_t iv_idx;
+	bool iv_update;
+
+	l_info("Node import method call");
+
+	node = node_new(uuid);
+	memset(&db_node, 0, sizeof(db_node));
+
+	net = node_get_net(node);
+	assert(net);
+
+	set_defaults(node);
+
+	if (!parse_imported_iv_index(jnode, &iv_idx, &iv_update)) {
+		l_error("Failed to parse imported iv idx");
+		goto fail;
+	}
+
+	if (!parse_imported_composition(jnode, &db_node)) {
+		l_error("Failed to parse composition data");
+		goto fail;
+	}
+
+	if (!parse_imported_unicast_addr(jnode, &unicast)) {
+		l_error("Failed to parse unicast address");
+		goto fail;
+	}
+
+	if (!parse_imported_device_key(jnode, &dev_key[0])) {
+		l_error("Failed to parse device key");
+		goto fail;
+	}
+
+	net_keys = parse_imported_net_key(jnode);
+
+	if (!net_keys) {
+		l_error("Failed to parse imported network key");
+		goto fail;
+	}
+
+	if (!parse_imported_elements(jnode, &db_node)) {
+		l_error("Failed to parse elements");
+		goto fail;
+	}
+
+	if (!parse_imported_finalize(&db_node, node)) {
+		l_error("Failed to finalize imported node");
+		goto fail;
+	}
+
+	if (!create_node_config(node)) {
+		l_error("Failed to create config");
+		goto fail;
+	}
+
+	if (!add_local_node(node, unicast, iv_update, iv_idx, dev_key,
+								net_keys)) {
+		l_error("Failed to add local node");
+		goto fail;
+	}
+
+	l_queue_foreach(net_keys, put_net_keyrings, node);
+
+	if (!keyring_put_remote_dev_key(node, unicast, node->num_ele, dev_key))
+		goto fail;
+
+	cb(user_data, MESH_ERROR_NONE, node);
+	return;
+
+fail:
+	cb(user_data, MESH_ERROR_FAILED, NULL);
+	json_object_put(jnode);
+
+	if (net_keys)
+		l_queue_destroy(net_keys, l_free);
+
+	if (node)
+		node_remove(node);
+}
+
 void node_create(const char *app_path, const char *sender, const uint8_t *uuid,
 					node_ready_func_t cb, void *user_data)
 {
 	struct managed_obj_request *req;
+	struct mesh_node *node;
```

Comment:
`mesh/node.c:2140:20: error: variable ‘node’ set but not used [-Werror=unused-but-set-variable]`